src/WebJobs.Script/Host/PrimaryHostCoordinator.cs

```diff
@@ -100,13 +102,19 @@ internal IDistributedLock LockHandle
                 throw new ArgumentOutOfRangeException(nameof(leaseTimeout), $"The {nameof(leaseTimeout)} should be between 15 and 60 seconds");
             }
 
-            var manager = new PrimaryHostCoordinator(lockManager, leaseTimeout, hostId, instanceId, traceWriter, loggerFactory, renewalInterval);
-            return manager;
+            // Dispose previously created instance to ensure releasing a lease before new instance is created.
+            if (latestCreatedInstance != null)
```

Comment:
This class shouldn't be holding onto instances and disposing them (anti-pattern). Instead we should be managing these instances.

To address the bug, it looks like your below disposal change is sufficient. That will prevent us from attempting to renew a lease we don't have, which was the error message reported. I asked some clarifying questions on the issue.

Reply:
@mathewc, the lease can be held by multiple hosts inside single instance as we use the same leaseId (instanceId) to acquire the lease. It's flow @brettsam mentioned:
Host1 -> acquire lock
Host1 -> renew lock
something triggers host recycle and creates a new host while disposing of the old host
Host2 -> acquire lock
Host1 -> release lock
Host2 -> renew lock -> throw

so we need to ensure:
Host1 -> release lock
Host2 -> acquire lock

We can use "ScriptHostManager" instead  but this will require new method in ScriptHost:
https://github.com/Azure/azure-webjobs-sdk-script/pull/2215

Reply:
I see what you're trying to do here -- only have 1 'active' instance of the PrimaryHostCoordinator at a time. But disposing it like this would be a surprise to anyone else using that instance and could break things elsewhere.

If we don't think that special `Dispose` method in #2215 will work 100% -- can we catch this 409 error upon 'renew', log it and try to re-acquire immediately?

Reply:
How it can be a surprise to anyone else? it just set up host as no primary.

The problem is we need to get rid of error logs (this was main concern of the original bug).